lisa/runners/lisa_runner.py

```diff
@@ -352,11 +352,20 @@ def _run_test_task(
         # on next test cases, close the connection after each test run. It will
         # be connected on the next command automatically.
         environment.nodes.close()
+        # Try to connect node(s), if cannot access node(s) of this environment,
+        # set the current environment as Bad. So that this environment won't be reused.
+        if not environment.nodes.test_connections():
+            environment.status = EnvironmentStatus.Bad
+            self._log.debug(
+                f"set environment '{environment.name}' as bad, "
+                f"because after test case '{test_result.name}', "
+                f"node(s) cannot be accessible."
+            )
+            environment.nodes.close()
```

Comment:
This line shouldn't be in the if clause. The connections should be closed, if the test returns True.

Reply:
updated